content/blog/2020/06/2020-06-17-external-fingerprint-storage.adoc

```diff
@@ -0,0 +1,160 @@
+tags:
+- plugins
+- fingerprint
+- cloud-native
+- external-storage
+- developer
+- redis
+- gsoc
+author: stellargo
+---
+
+= External Fingerprint Storage Phase-1 Updates
+
+Externalizing fingerprint storage is a Jenkins project in GSoC 2020.
+We are working on building a pluggable storage engine for supporting plugin configurable external storages for Jenkins (see jep:226[]).
+Along with this API, we are also working on a link:https://github.com/jenkinsci/redis-fingerprint-storage-plugin[reference implementation] in the form of a plugin, powered by Redis.
+
+As phase 1 of this project comes to an end, this blog post serves as a summary of the progress we made to the entire Jenkins community.
+
+== Current State
+
+* The link:https://github.com/jenkinsci/jenkins/pull/4731[new API] introduced in Jenkins core is under review. Once merged, it will offer developers to extend it to build external fingerprint storage plugins.
+* The link:https://github.com/jenkinsci/redis-fingerprint-storage-plugin[Redis Fingerprint Storage Plugin] is alpha release ready. We would immensely appreciate any feedback.
+
+== Introducing the new API for plugin developers
+
+With link:https://github.com/jenkinsci/jenkins/pull/4731[PR-4731], we introduce a new fingerprint storage API, allowing configuring custom storage engines.
+We exposed the following methods in the new `FingerprintStorage` class:
+
+* `void save()`
+
+** Saves the given Fingerprint in the storage.
+
+* `Fingerprint load(String id)`
+
+** Returns the Fingerprint with the given unique ID. The unique ID for a fingerprint is defined by Fingerprint#getHashString().
+
+* `void delete(String id)`
+
+** Deletes the Fingerprint with the given unique ID.
+
+* `boolean isReady()`
+
+** Returns true if there's some data in the fingerprint database corresponding to the particular Jenkins instance.
+
+
+== Introducing Redis Fingerprint Storage Plugin
+
+Redis Fingerprint Storage Plugin uses the new External Fingerprint Storage API to store the fingerprints in a Redis instance.
+
+=== Installation:
+
+Clone the plugin source code to your local machine:
+
+[source, bash]
+----
+git clone https://github.com/jenkinsci/redis-fingerprint-storage-plugin
+cd redis-fingerprint-storage-plugin
+----
+
+Install the plugin:
+
+```
+mvn clean install
+```
+
+(Note: Use `-DskipTests` to skip tests)
+
+Next, Run the plugin:
+
+[source, bash]
+----
+mvn hpi:run
+----
+
+By default this runs a Jenkins instance on localhost:8080.
+You may provide a custom port number using `-Djetty.port=<port>`
+
+If you want to test it with your Jenkins server, after `mvn clean install` follow these steps in your Jenkins instance:
+
+. Select `Manage Jenkins`
+
+. Select `Manage Plugins`
+
+. Go to the `Advanced` tab
+
+. Under `Upload Plugin` section, select `Choose file`
+
+. Select `<path_to_plugin>/plugin/target/redis-fingerprint-storage.hpi`
+
+. Select `Upload`
+
+. Select `Restart Jenkins when installation is complete and no jobs are running`
```

Comment:
I think it can be replaced by an alpha release and Experimental update center, right?
https://www.jenkins.io/doc/developer/publishing/releasing-experimental-updates/

Reply:
Ill update it once we have an alpha release, or should i do it in advance?

Reply:
If you want to post it before the Alpha release, we can keep it as is